src/platform/common.c

```diff
@@ -0,0 +1,51 @@
+#include <windows.h>
+#include "core.h"
+
+u8 REG_BASE[IO_SIZE] = { 0 };
+u16 INTR_CHECK = 0;
+IntrFunc INTR_VECTOR = IntrMain;
+
+#define CHECK_INTR(res, input, flags)                                                   \
```

Comment:
Minor, but I would just define this in the body of `IntrMain` and unset it at the end since it only works for that function?